msal/region.py

```diff
@@ -0,0 +1,33 @@
+import os
+import json
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+def _detect_region(http_client=None):
+    region = _detect_region_of_azure_function()  # It is cheap, so we do it always
+    if http_client and not region:
```

Comment:
>http_client [](http://example.com/codeflow?start=7&length=11)

I recommend a dedicated http client for this call with a timeout set to 20 seconds or so... Neha and Bogdan can advise on the value to use.

Reply:
Actually, the [internal proposal](https://identitydivision.visualstudio.com/DevEx/_git/AuthLibrariesApiReview?path=%2FPinAuthToRegion%2FAAD%20SDK%20Proposal%20to%20Pin%20Auth%20to%20region.md&_a=preview&anchor=goals) suggests a 2-second timeout only. But I already had an off-line conversation with Bogdan days ago about the situation that, in Python there is no a system-wide `System.Net.Http.HttpClient`, so, there is no universal way for MSAL Python to configure timeout, especially when the incoming `http_client` is provided by app developer.

Currently, we try to provide clear documentation, also included in this PR.

The conversation did confirm one thing, that the region detection would be turned off by default.

Reply:
Right, I meant 2 seconds... There must be a way even for Python to ensure this. 

Region detection should be used in the context where region detection is needed, always, not off by default. Azure region is not the default behavior of AquireTokenForClient (though it may be some day) if that is what you mean.